compiler-plugin/src/main/scala/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -1326,10 +1326,15 @@ class ProtobufGenerator(
         val clearMethod = "clear" + field.upperScalaName
         val singleType  = field.singleScalaTypeName
         printer
-          .when(field.supportsPresence || field.isInOneof) { p =>
+          .when(
+            (field.supportsPresence || field.isInOneof) && (message.generateGetters || message.generateLenses)
+          ) { p =>
             val default = defaultValueForGet(field)
+            val scope =
+              if (message.generateGetters) ""
+              else s"private[${message.getFile.scalaPackage.xs.last}] "
```

Comment:
Not sure if there's a prettier way to access the innermost sub-package!

Reply:
Can this be private and not package-private?

Reply:
You could try `private[this]` if necessary.

Reply:
`private[this]` does not work because the access in the lens is done in the companion object.
But you're right a plain simple `private` does work! Looks much cleaner!